modules/sdk-coin-tia/src/lib/transactionBuilder.ts

```diff
@@ -0,0 +1,153 @@
+import { BaseAddress, BuildTransactionError, InvalidTransactionError, TransactionType } from '@bitgo/sdk-core';
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { Secp256k1, sha256 } from '@cosmjs/crypto';
+import { makeSignBytes } from '@cosmjs/proto-signing';
+
+import {
+  CosmosTransaction,
+  DelegateOrUndelegeteMessage,
+  FeeData,
+  SendMessage,
+  WithdrawDelegatorRewardsMessage,
+  CosmosTransactionBuilder,
+} from '@bitgo/abstract-cosmos';
+
+import { TiaTransaction } from './transaction';
+import utils from './utils';
+
+export abstract class TiaTransactionBuilder extends CosmosTransactionBuilder {
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+    this._transaction = new TiaTransaction(_coinConfig);
+  }
+
+  /**
+   * Sets messages to the transaction body. Message type will be different based on the transaction type
+   * - For @see TransactionType.StakingActivate required type is @see DelegateOrUndelegeteMessage
+   * - For @see TransactionType.StakingDeactivate required type is @see DelegateOrUndelegeteMessage
+   * - For @see TransactionType.Send required type is @see SendMessage
+   * - For @see TransactionType.StakingWithdraw required type is @see WithdrawDelegatorRewardsMessage
+   * @param {(SendMessage | DelegateOrUndelegeteMessage | WithdrawDelegatorRewardsMessage)[]} messages
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  abstract messages(messages: (SendMessage | DelegateOrUndelegeteMessage | WithdrawDelegatorRewardsMessage)[]): this;
+
+  /**
+   * The transaction type.
+   */
+  protected abstract get transactionType(): TransactionType;
+
+  /**
+   * Sets gas budget of this transaction
+   * Gas budget consist of fee amount and gas limit. Division feeAmount/gasLimit represents
+   * the gas-fee and it should be more than minimum required gas-fee to process the transaction
+   * @param {FeeData} gasBudget
+   * @returns {TransactionBuilder} this transaction builder
+   */
+  gasBudget(gasBudget: FeeData): this {
+    utils.validateGasBudget(gasBudget);
+    this._gasBudget = gasBudget;
+    return this;
+  }
+
+  /**
+   * Initialize the transaction builder fields using the decoded transaction data
+   * @param {TiaTransaction} tx the transaction data
+   */
+  initBuilder(tx: TiaTransaction): void {
```

Comment:
seems like a method which can be abstracted into the parent builder class. We can pass in the `tx`(of type TiaTransaction) to the parent builderclass, which takes in any `CosmosTransaction` and initializes the builder

Reply:
There are few methods that uses validateAmount() and isValidAddress() methods which are coin specific. And hence I had to keep them in their sdks, rest all methods are abstracted into abstract-cosmos sdk.